### ISSUE TITLE
Error thrown in Rmd in document chunk output by `knitr::include_url` and `knitr::include_app` 

### ISSUE THREAD

**Issue by cderv** [rstudio/rstudio#6476]
Issue first reported in https://github.com/yihui/knitr/issues/1820

### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.911
    OS Version      : Windows 10
    R Version       : 3.6.3

### Steps to reproduce the problem

* Paste this in a new Rmd document 
````r
---
title: "test"
output: html_document
editor_options: 
  chunk_output_type: inline
---

```{r}
knitr::include_url("https://rstudio.com")
```

```{r}
knitr::include_app("https://yihui.shinyapps.io/miniUI/")
```
````
Try to run by chunk using green arrow Run Current chunk. You'll get an error. 
![image](https://user-images.githubusercontent.com/6791940/76711417-1fddf900-6710-11ea-868b-b76785ce2e83.png)

If chunk output is in console, by changing the yaml header, no issue. 
```yaml
editor_options: 
  chunk_output_type: console
```

Let's note that the document renders fine when knitted.

### Describe the problem in detail

`knitr::include_url` and `knitr::include_app` will create special object with class `knit_embed_url` and `knit_asis`
```r
> knitr::include_url("https://rstudio.com")
$url
[1] "https://rstudio.com"

$height
[1] "400px"

$url.orig
[1] "https://rstudio.com"

attr(,"class")
[1] "knit_embed_url" "knit_asis"   
```

It seems that RStudio handle this object using `print.knit_asis` only from here: 
https://github.com/rstudio/rstudio/blob/28a5e3ff51462e836e4cd002507853ec92de87ee/src/cpp/session/modules/NotebookHtmlWidgets.R#L278
https://github.com/rstudio/rstudio/blob/28a5e3ff51462e836e4cd002507853ec92de87ee/src/cpp/session/modules/NotebookHtmlWidgets.R#L66-L76

The error seems to be thrown from there, because `x` is a list, and `enc2utf8` can't handle it.

### Describe the behavior you expected

I would expect RStudio to have a special `print.knit_embed_url` that deals with this output. Either by including as in HTML output by an iframe, or at list having a message and the same result as in console. Anything but an error.

Please tell me if I am wrong with my diagnosis of RStudio IDE not dealing correctly with this special type of output. I may have misunderstood the problem on this one. 🤔  Thank you ! 

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by chrisaberson** [comment]
I am having this exact issue using include_app

**Comment by systemnova** [comment]
I'm also having this issue.

**Comment by cderv** [comment]
This issue was closed by PR closing keyword from yihui/knitr because we merged a fix there that solve this. 

Basically, `include_app()` and `include_url()` will now return an object of class `c("knit_embed_url", "knit_asis_url") ` and RStudio IDE won't have a special hook to handle it. This means the default print method will apply interactively and it will no more error. but show the list as is: 

![](https://user-images.githubusercontent.com/6791940/121547327-56d2b580-ca0c-11eb-8a4b-3ac12249e2d3.png)

This could be improved if `print.knit_embed_url` had a special handling but it seems enough to fix this issue quickly for R user in next knitr release. 

I don't think the `print.knit_asis` IDE's hook will have other edge case but we'll see that in the future. The IDE should avoid `enc2utf8()` is the object if not a character vector.
